### ISSUE TITLE
"Failed to decompress the IDAT stream" on the TI 84 Plus CE

### ISSUE THREAD

**Issue by TheMiningTeamYT** [kikuchan/pngle#12]
I'm attempting to use PNGLE to add PNG support to my USB image viewer for the TI 84 Plus CE, BMP84CE. However, after calling `pngle_feed` I'm getting the error `Failed to decompress the IDAT stream`. From debug_printf, I'm also getting the message `[pngle] tinfl_decompress() failed with status -1!` The code of tinfl_decompress is too dense for me to read, so I was hoping I could get some help. You can see my code [here.](https://github.com/TheMiningTeamYT/Bitmap84CE/tree/png) (in particular, I'm invoking PNGLE [here](https://github.com/TheMiningTeamYT/Bitmap84CE/blob/png/src/png.cpp#L186C1-L186C63).)
FYI, the TI 84 Plus CE uses a 24 bit (yes, _24 bit!_) eZ80, meaning `int` and `unsigned int` are 24 bits wide, not 32 bits wide. I already had to fix the `mz_uint32` types in miniz, and the `read_uint32` function in PNGLE because of this. Hope this helps you figure out what's going wrong. If you have any questions, please ask.

**Comment by TheMiningTeamYT** [comment]
Okay, I figured out what's going on.
The TI 84 Plus CE toolchain compiler doesn't like this macro: https://github.com/kikuchan/pngle/blob/8bf3862f84a4643e75a7d92e046d7c07f934cead/src/miniz.c#L1382
It was incrementing the pointer more than it should, which was causing the ZLIB header to get read incorrectly.
I changed it to this: https://github.com/TheMiningTeamYT/Bitmap84CE/blob/74a767bcd400b4a32cf7cbd153d931b57e1abf44/src/pngle/miniz.c#L1393
and it started working. (This is also easier to read IMO.)
